OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -305,41 +457,123 @@ public string VoicePhraseForOrder(Actor self, Order order)
 				case "Enter":
 				case "ReturnToBase":
 				case "Stop":
-					return info.Voice;
+					return Info.Voice;
 				default: return null;
 			}
 		}
-	}
 
-	class AircraftMoveOrderTargeter : IOrderTargeter
-	{
-		public string OrderID { get { return "Move"; } }
-		public int OrderPriority { get { return 4; } }
-		public bool OverrideSelection { get { return false; } }
+		public void ResolveOrder(Actor self, Order order)
+		{
+			if (order.OrderString == "Move")
+			{
+				var cell = self.World.Map.Clamp(order.TargetLocation);
+				var explored = self.Owner.Shroud.IsExplored(cell);
 
-		readonly AircraftInfo info;
+				if (!explored && !Info.MoveIntoShroud)
+					return;
 
-		public AircraftMoveOrderTargeter(AircraftInfo info) { this.info = info; }
+				if (!order.Queued)
+					UnReserve();
 
-		public bool CanTarget(Actor self, Target target, List<Actor> othersAtTarget, TargetModifiers modifiers, ref string cursor)
-		{
-			if (target.Type != TargetType.Terrain)
-				return false;
+				var target = Target.FromCell(self.World, cell);
+
+				self.SetTargetLine(target, Color.Green);
+
+				if (isPlane)
+					self.QueueActivity(order.Queued, new FlyAndContinueWithCirclesWhenIdle(self, target));
+				else
+					self.QueueActivity(order.Queued, new HeliFlyAndLandWhenIdle(self, target, Info));
+			}
+			else if (order.OrderString == "Enter")
+			{
+				if (!order.Queued)
+					UnReserve();
+
+				if (Reservable.IsReserved(order.TargetActor))
+				{
+					if (isPlane)
+						self.QueueActivity(new ReturnToBase(self));
+					else
+						self.QueueActivity(new HeliReturnToBase(self));
+				}
+				else
+				{
+					self.SetTargetLine(Target.FromActor(order.TargetActor), Color.Green);
+
+					if (isPlane)
+					{
+						self.QueueActivity(order.Queued, Util.SequenceActivities(
+							new ReturnToBase(self, order.TargetActor),
+							new ResupplyAircraft(self)));
+					}
+					else
+					{
+						Action enter = () =>
+						{
+							var exit = order.TargetActor.Info.Traits.WithInterface<ExitInfo>().FirstOrDefault();
+							var offset = (exit != null) ? exit.SpawnOffset : WVec.Zero;
+
+							self.QueueActivity(new HeliFly(self, Target.FromPos(order.TargetActor.CenterPosition + offset)));
+							self.QueueActivity(new Turn(self, Info.InitialFacing));
+							self.QueueActivity(new HeliLand(self, false));
+							self.QueueActivity(new ResupplyAircraft(self));
+							self.QueueActivity(new TakeOff(self));
+						};
+
+						self.QueueActivity(order.Queued, new CallFunc(enter));
+					}
+				}
+			}
+			else if (order.OrderString == "Stop")
+			{
+				self.CancelActivity();
+				if (GetActorBelow() != null)
+				{
+					self.QueueActivity(new ResupplyAircraft(self));
+					return;
```

Comment:
Can't we just use
```
if (GetActorBelow() != null)
	return;
```
straight at the beginning instead of throwing away the activity and creating a new one?

Also what about checking if `Reservation` is `null` instead of using `GetActorBelow`?